File: src/main/java/net/frozenblock/configurableeverything/entity/util/skeleton/ai/SkeletonFleeSunGoal.java
package net.frozenblock.configurableeverything.entity.util.skeleton.ai;

import net.frozenblock.configurableeverything.config.EntityConfig;
import net.frozenblock.configurableeverything.config.MainConfig;
import net.minecraft.world.entity.PathfinderMob;
import net.minecraft.world.entity.ai.goal.FleeSunGoal;

public class SkeletonFleeSunGoal extends FleeSunGoal {

	public SkeletonFleeSunGoal(PathfinderMob mob, double speedModifier) {
		super(mob, speedModifier);
	}

	@Override
	public boolean canUse() {
		var entity = MainConfig.get(false).entity;
		var skeleton = EntityConfig.get(false).skeleton;
		// null safety in java moment
		if (entity != true || skeleton == null || skeleton.skeletonsAvoidSun != true) {
			return false;
		}
		return super.canUse();
	}
}

File: src/main/java/net/frozenblock/configurableeverything/entity/util/skeleton/ai/SkeletonFleeSunGoal.kt
package net.frozenblock.configurableeverything.entity.util.skeleton.ai

import net.frozenblock.configurableeverything.config.EntityConfig
import net.frozenblock.configurableeverything.config.MainConfig
import net.minecraft.world.entity.PathfinderMob
import net.minecraft.world.entity.ai.goal.FleeSunGoal

class SkeletonFleeSunGoal(mob: PathfinderMob, speedModifier: Double) : FleeSunGoal(mob, speedModifier) {
    override fun canUse(): Boolean {
        val config = EntityConfig.get()

        if (!MainConfig.entity.get())
            return super.canUse()

        return config.skeleton.skeletonsAvoidSun && super.canUse()
    }
}
